src/Pos/Block/Core/Main/Misc.hs

```diff
@@ -4,15 +4,18 @@
 -- | Miscellaneous instances, etc. Related to the main blockchain of course.
 
 module Pos.Block.Core.Main.Misc
```

Comment:
OMG I don't think it's a proper naming. Why put important creation helpers into `Misc`? You could leave it as `Pure`, couldn't you?